compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -327,7 +327,9 @@ fn check_for_bindings_named_same_as_variants(
                 if let ty::Adt(edef, _) = pat_ty.kind() {
                     if edef.is_enum()
                         && edef.variants.iter().any(|variant| {
-                            variant.ident == ident && variant.ctor_kind == CtorKind::Const
+                            // Compare symbols - this is a lint, so we don't care
+                            // about the span
+                            variant.name == ident.name && variant.ctor_kind == CtorKind::Const
```

Comment:
Could this create a false positive if both have the same name but hygiene makes them invisible to each other?

Reply:
Good catch - this should be using `.ident(tcx)` instead